test/__snapshots__/test.js.snap

```diff
@@ -395,6 +395,173 @@ const Impl = require(\\"../implementations/Factory.js\\");
 "
 `;
 
+exports[`HTMLConstructor.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+const impl = utils.implSymbol;
+
+function HTMLConstructor() {
+  iface.setup(this, [], { newTarget: new.target });
```

Comment:
The constructor could be improved by:

1. implementing the following step in the spec:

   > If NewTarget is equal to the active function object, then throw a TypeError and abort these steps.

2. throwing in case `new.target` is undefined

   > When invoked with [[Call]], the NewTarget value is undefined, and so the algorithm below will immediately throw. 

I.e.

```js
if (new.target === undefined) {
  throw new TypeError("Failed to construct 'HTMLConstructor'. Please use the 'new' operator; this constructor " +
                      "cannot be called as a function.");
}
if (new.target === HTMLConstructor) {
  throw new TypeError("Illegal constructor");
}
```